src/model.py

```diff
@@ -855,6 +849,46 @@ def init_optimizer(self):
                 verbose=True,
                 eps=1e-10,
             )
+
+        # make sure non-single-task-specific components are added to optimizers
+        if other_components is not None:
+            for component in other_components:
+                if other_components[component] is not {}:
+                    lr = other_components[component].get("lr")
+                    lr = lr if lr is not None else self.learning_rate
+
+                    # IMN architecture
+                    if component == "shared":
+                        tasks = other_components[component].get("tasks")
+                        tasks = tasks if tasks is not None else []
+
+                        for task in tasks:
+                            optimizers[task].add_param_group(
+                                {"params": self.components[component].parameters(), "lr":lr}
+                            )
+                   
+                    # RACL architecture
+                    elif component == "relations":
+                        for layer in self.components["relations"]:
+                            for task in self.subtasks:
+                                if task in layer:
+                                    print("adding {} to optimizer {}".format(layer, task))
+                                    optimizers[task].add_param_group(
+                                        {"params": self.components[component][layer].parameters(), "lr":lr}
+                                    )
+
+                                    # only passes this if when layer=shared_at_polarity
+                                    if "shared" in layer:
+                                        print("adding {} to optimizer {}".format(layer, task))
+
+                                        optimizers["target"].add_param_group(
+                                            {"params": self.components[component][layer].parameters(), "lr":lr}
+                                        )
+                                        optimizers["expression"].add_param_group(
+                                            {"params": self.components[component][layer].parameters(), "lr":lr}
+                                        )
```

Comment:
Fixed in reference to this conversation: https://github.com/pmhalvor/fgsa/pull/36#discussion_r836540597